### PR TITLE
Minor fixes for Clang compilation

### PR DESCRIPTION
**Pull request by henryymliu**
Also added CMake option for Visual Studio Code Intellisense.

### CODE REVIEW

File: src/thunderbots/CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,9 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
     set_property(CACHE CMAKE_BUILD_TYPE PROPERTY VALUE "Release")
 endif()
 
+# For Visual Studio Code Intellisense
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```
Comment (garethellis0) [inline src/thunderbots/CMakeLists.txt]: :+1: this works for vim too! Might want to change comment to something like `# For external editor code completion and hinting (Visual Studio Code, Vim, etc.)`

**Review by garethellis0 — COMMENTED** [review]
Is this all we have to do to get this compiling with clang? Is this compiling with clang now? (Not seeing changes I would expect in the `CMakeLists.txt`)

File: src/thunderbots/CMakeLists.txt (hunk 1 above)
Comment (henryymliu) [reply src/thunderbots/CMakeLists.txt]: Ah, didn't know Vim (well, at least YCM) uses this .json too. Will update.

**Comment by henryymliu** [comment]
Wasn't too clear in the review request, but the intention here is not to switch to Clang straight away. On my machine at home I was able to use Clang for compilation by changing the CMake environment variables.

**Comment by henryymliu** [comment]
`compile_commands.json` is generated in the `build/` folder, which is already ignored.

**Review by garethellis0 — APPROVED** [review]

**Comment by garethellis0** [comment]
OH whoops, sorry, accidentally merged this. Let me know if it's an issue and I can revert it. Sorry!

**Pull request merged**